app.js

```diff
@@ -54,6 +59,38 @@ app.use(function(req, res, next) {
    }
 });
 
+app.use(cookieParser());
+app.use(session({
+        secret:'secret_token',
+        cookie: {maxAge: 24*3600*1000},//cookie max age set to one day
+        resave: true,
+        saveUninitialized: true
+        }));
+
+//if username and password supplied enabled auth
+var seedUsername = nconf.get('USERNAME')
+var seedPassword = nconf.get('PASSWORD')
+if(seedUsername && seedUsername !== "" && seedPassword && seedPassword !== "") {
```

Comment:
If `USERNAME` or `PASSWORD` is set, but not both, throw an exception. This is akin to the logic you added for HTTPS: I might be trying to set the login, but mess one up and expect things to be secure, when in reality they're not. (I just did this by misspelling PASSWORD on the command line in fact.)